File: ecommerce-spring-main/src/main/java/com/ecommerce/service/ReplayCommentService.java
package com.ecommerce.service;

import java.util.List;

import com.ecommerce.modal.ReplayComment;

public interface ReplayCommentService {
   ReplayComment addReplayCommentToComment(ReplayComment rcomment, long idComment);
		
   ReplayComment findReplayCommentById(long id);
	
	void deleteReplayComment(long id);
	
	List<ReplayComment> findReplayCommentsForComment(long idComment);
	
	List<ReplayComment> findAllReplayComments();	
	

}

File: Eshopper/src/app/Services/replay-comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ReplayComment } from '../models/model';

@Injectable({
  providedIn: 'root'
})
export class ReplayCommentService {
 
  constructor(private http: HttpClient) { }
  addReplayCommentToComment(recomment : ReplayComment,idComment: any) : Observable<ReplayComment> {
    return this.http.post<ReplayComment>(`http://localhost:8080/api/addReplayCommentToComment/${idComment}`,recomment);
  }
  findReplayCommentsForComment(idComment: any): Observable<ReplayComment[]> {
    return this.http.get<ReplayComment[]>(`http://localhost:8080/api/findReplayCommentsForComment/${idComment}`);
  }

  findAllReplayComments(): Observable<ReplayComment[]> {
    return this.http.get<ReplayComment[]>(`http://localhost:8080/api/findAllReplayComments`);
  }
 

}
 

